runtime/executor/src/main/java/org/apache/nemo/runtime/executor/task/TaskExecutor.java

```diff
@@ -478,17 +511,24 @@ private boolean handleDataFetchers(final List<DataFetcher> fetchers) {
     return map;
   }
 
-  private Map<String, List<OperatorVertex>> getInternalAdditionalOutputMap(
+  private Map<String, List<NextOperatorInfo>> getInternalAdditionalOutputMap(
```

Comment:
Maybe merge `getInternalAdditionalOutputMap` and `getInternalMainOutputs`, and remove the intermediate`edgeIndexMap`?

Pair<internal, additional> merged() {
  internal = new;
  additional = new;

  for (outedge) {
    if ... 
    then internal.add(index, ...) 
    else additional.add(index, ...)
  }
}

Reply:
https://issues.apache.org/jira/browse/NEMO-253